pkg/virt-api/rest/subresource.go

```diff
@@ -90,7 +90,7 @@ func NewSubresourceAPIApp(virtCli kubecli.KubevirtClient, consoleServerPort int,
 	// the virtCli is nil, and accessing GeneratedKubeVirtClient() would cause nil dereference.
 	var instancetypeMethods instancetype.Methods
 	if virtCli != nil {
-		instancetypeMethods = instancetype.NewMethods(virtCli)
+		instancetypeMethods = instancetype.NewMethods(nil, nil, nil, nil, nil, virtCli)
```

Comment:
This looks really weird to me. Can't the struct be used directly? Is there a benefit in using the `NewMethods` func?

Reply:
Apologies missed this, `methods` isn't public but that's something we could change so I really don't know tbh.

Reply:
Can you investigate if this would be a lot of effort? Is there a benefit to hiding the actual struct?